src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -101,7 +114,15 @@ public Book update(@RequestBody Book book, @ApiParam(value = "book id", required
     }
 
     @GetMapping
-    public List<Book> find(@RequestParam(required = false) String publisher, @RequestParam(required = false) String genre, @RequestParam(required = false) String year) {
-        return bookRepository.findByPublisherAndGenreAndYear(publisher,genre,year);
+    public List<Book> getAll(@RequestParam(required = false) String genre, @RequestParam(required = false) String author, @RequestParam(required = false) String title,
+                             @RequestParam(required = false) String image, @RequestParam(required = false) String subtitle, @RequestParam(required = false) String publisher,
+                             @RequestParam(required = false) String year, @RequestParam(required = false) String isbn, @RequestParam(defaultValue = "0") int page,
+                             @RequestParam(defaultValue = "id") String sort ) throws BadSortException {
+        if(!permittedSort.contains(sort))
+            throw new BadSortException();
+        Pageable pageable =
+                PageRequest.of(page, 10, Sort.by(sort));
+         return bookRepository.
+                findAll(genre,author,title,image,subtitle,publisher,year,isbn, pageable);
```

Comment:
You could set the bookdepository object, directly as follows: bookRepository.findAll(genre,author,title,image,subtitle,publisher,year,isbn, PageRequest.of(page, 10, Sort.by(sort)));

In line 136 of userRepository, you could do the same.